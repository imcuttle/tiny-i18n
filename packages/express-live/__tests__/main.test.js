/**
 * @file main
 * @author Cuttle Cong
 * @date 2018/6/21
 * @description
 */
const request = require('supertest')
const express = require('express')
const { join } = require('path')

const live = require('../')

function makeFixture(name) {
  return join(__dirname, 'fixture', name)
}

describe('express-live', function() {

  describe('simple', function () {
    const app = express()
    app.use('/i18n/', live(makeFixture('simple')))

    it('should main languages', done => {
      request(app)
        .get('/i18n/languages')
        .end((err, res) => {
          expect(res.body).toEqual(
            expect.objectContaining({
              data: ['en-US', 'zh-CN']
            })
          )

          done()
        })
    })

    it('should main word', done => {
      request(app)
        .get('/i18n/word?key=create&lang=en-US')
        .end((err, res) => {
          expect(res.body).toEqual(
            expect.objectContaining({
              code: 200,
              data: '创建'
            })
          )

          done()
        })
    })

    it('should main update', done => {
      request(app)
        .post('/i18n/update')
        .send({ key: 'lalala', value: 'xixixi', lang: 'zh-CN' })
        .end((err, res) => {
          expect(res.body).toEqual(
            expect.objectContaining({ code: 502, data: 'Error: [tiny-i18n fs] The value of `lalala` equals `xixixi`' })
          )

          done()
        })
    })
  });

  describe('comple', function () {
    const app = express()
    app.use('/i18n/', live(makeFixture('complex')))

    it('should main languages', done => {
      request(app)
        .get('/i18n/languages')
        .end((err, res) => {
          expect(res.body).toEqual(
            expect.objectContaining({
              data: ['en-us', 'zh-cn']
            })
          )

          done()
        })
    })

    it('should main word', done => {
      request(app)
        .get('/i18n/word?key=create&lang=en-us/common')
        .end((err, res) => {
          expect(res.body).toEqual(
            expect.objectContaining({
              code: 200,
              data: '创建'
            })
          )

          done()
        })
    })

    it('should main update error', done => {
      request(app)
        .post('/i18n/update')
        .send({ key: 'same_value', value: 'xixixi', lang: 'zh-cn/main' })
        .end((err, res) => {
          expect(res.body).toEqual(
            expect.objectContaining({ code: 502, data: 'Error: [tiny-i18n fs] The value of `same_value` equals `xixixi`' })
          )

          done()
        })
    })

    it('should main update', done => {
      request(app)
        .post('/i18n/update')
        .send({ key: 'site_title', value: Date.now(), lang: 'zh-cn/main' })
        .end((err, res) => {
          expect(res.body).toMatchSnapshot()
          done()
        })
    })

    it('should main update error2', done => {
      request(app)
        .post('/i18n/update')
        .send({ key: 'site_title', value: Date.now(), lang: 'zh-cn/mainx' })
        .end((err, res) => {
          expect(res.body).toMatchSnapshot()
          done()
        })
    })
  });

})
