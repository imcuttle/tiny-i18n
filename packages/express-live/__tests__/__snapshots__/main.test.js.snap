// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`express-live comple should main update 1`] = `
Object {
  "code": 200,
  "data": true,
}
`;

exports[`express-live comple should main update error2 1`] = `
Object {
  "code": 502,
  "data": "Error: [tiny-i18n fs] The language: zh-cn/mainx is not found.",
}
`;
